PebbleSharp.Core/Install/InstallClient.cs

```diff
@@ -0,0 +1,207 @@
+using PebbleSharp.Core.BlobDB;
+
+namespace PebbleSharp.Core.Install
+{
+	public class InstallClient
+	{
+		private Pebble _pebble;
+		public InstallClient(Pebble pebble)
+		{
+			_pebble = pebble;
+		}
+		public async Task InstallAppAsync( AppBundle bundle, IProgress<ProgressValue> progress = null )
+		{
+
+			string version = _pebble.Firmware.Version;
+			version = version.Replace("v", "");
+			var components = version.Split(new char[] { '.','-' }, StringSplitOptions.RemoveEmptyEntries);
+
+			int i;
+			IList<int> versionComponents = components.Where(x=>int.TryParse(x,out i)).Select(x => int.Parse(x)).ToList();
+			if (versionComponents[0] < 3) 
+			{
+				await InstallAppLegacyV2 (bundle, progress);
+			} 
+			else 
+			{
+				await InstallAppAsyncV3 (bundle, progress);
+			}
+		}
+
+
+		private async Task InstallAppAsyncV3(AppBundle bundle,IProgress<ProgressValue> progress)
+		{
+			//https://github.com/pebble/libpebble2/blob/master/libpebble2/services/install.py
+
+			var meta = AppMetaData.FromAppBundle(bundle);
+
+			var bytes = meta.GetBytes();
+			var result = await _pebble.BlobDBClient.Delete(BlobDatabase.App, meta.UUID.Data);
+
+			result = await _pebble.BlobDBClient.Insert(BlobDatabase.App, meta.UUID.Data, bytes);
+
+			if (result.Response == BlobStatus.Success)
+			{
+				var startPacket = new AppRunStatePacket();
+				startPacket.Command = AppRunState.Start;
+				startPacket.UUID = meta.UUID;
+				//app_fetch = self._pebble.send_and_read(AppRunState(data=AppRunStateStart(uuid=app_uuid)), AppFetchRequest)
+
+				var runStateResult = await _pebble.SendMessageAsync<AppFetchRequestPacket>(Endpoint.AppRunState, startPacket.GetBytes());
+
+				if (!runStateResult.Success)
+				{
+					throw new InvalidOperationException("Pebble replied invalid run state");
+				}
+
+				if (!meta.UUID.Equals(runStateResult.UUID))
+				{
+					var response = new AppFetchResponsePacket();
+					response.Response = AppFetchStatus.InvalidUUID;
+					await _pebble.SendMessageNoResponseAsync(Endpoint.AppFetch, response.GetBytes());
+					throw new InvalidOperationException("The pebble requested the wrong UUID");
+				}
+
+				var putBytesResponse = await _pebble.PutBytesClient.PutBytes(bundle.App, TransferType.Binary, appInstallId:(uint)runStateResult.AppId);
+				if (!putBytesResponse)
+				{
+					throw new InvalidOperationException("Putbytes failed");
+				}
+
+				if (bundle.HasResources)
+				{
+					putBytesResponse = await _pebble.PutBytesClient.PutBytes(bundle.Resources, TransferType.Resources, appInstallId:(uint)runStateResult.AppId);
+					if (!putBytesResponse)
+					{
+						throw new InvalidOperationException("Putbytes failed");
+					}
+				}
+
+				//TODO: add worker to manifest and transfer it if necassary
+				//if (bundle.HasWorker)
+				//{
+				//await PutBytesV3(bundle.Worker, TransferType.Worker, runStateResult.AppId);
+				//}
+
+			}
+			else 
+			{
+				throw new DataMisalignedException("BlobDB Insert Failed");
+			}
+		}
```

Comment:
In this method there are several cases that throw InvalidOperationExceptions. I would suggest using PebbleException or a derived class in place of them.